llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -4718,14 +4718,14 @@ class SearchOptions {
     const MachineInstr *MI = MRI.getVRegDef(Reg);
     unsigned Opc = MI->getOpcode();
 
-    if (Opc < TargetOpcode::GENERIC_OP_END) {
-      // Keep same for generic op.
-      HasNeg = true;
-    } else if (Opc == TargetOpcode::G_INTRINSIC) {
+    if (Opc == TargetOpcode::G_INTRINSIC) {
       Intrinsic::ID IntrinsicID = cast<GIntrinsic>(*MI).getIntrinsicID();
       // Only float point intrinsic has neg & neg_hi bits.
       if (IntrinsicID == Intrinsic::amdgcn_fdot2)
         HasNeg = true;
+    } else if (Opc < TargetOpcode::GENERIC_OP_END) {
```

Comment:
Should probably use TargetInstrInfo::isGenericOpcode? 

If G_INTRINSIC doesn't count as isGenericOpcode, should fix that

Reply:
Thanks, done.